### PR TITLE
[Doc] Update User Privilege Examples

### PR DESCRIPTION
**Pull request by EsoragotoSpirit**
Signed-off-by: EsoragotoSpirit <wanglichen@starrocks.com>

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes:

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/administration/User_privilege.md [hunk 1]
```diff
@@ -420,15 +420,15 @@ We recommend you customize roles to manage privileges and users. The following e
...
 
    ```SQL
-   -- Create a role.
+   --Create a role.
    CREATE ROLE read_only;
-   -- Grant USAGE privileges on all catalogs to the role.
+   --Grant USAGE privileges on all catalogs to the role.
    GRANT USAGE ON ALL CATALOGS TO ROLE read_only;
-   -- Grant SELECT privileges on all tables to the role.
+   --Grant SELECT privileges on all tables to the role.
```
Comment (evelynzhaojie) [inline docs/administration/User_privilege.md:427]: "on all tables in all databases"?

File: docs/administration/User_privilege.md [hunk 2]
```diff
@@ -420,15 +420,15 @@ We recommend you customize roles to manage privileges and users. The following e
...
    GRANT SELECT ON ALL TABLES IN ALL DATABASES TO ROLE read_only;
-   -- Grant SELECT privileges on all views to the role.
+   --Grant SELECT privileges on all views to the role.
    GRANT SELECT ON ALL VIEWS IN ALL DATABASES TO ROLE read_only;
-   -- Grant SELECT privileges on all materialized views and the privilege to accelerate queries with them to the role.
+   --Grant SELECT privileges on all materialized views and the privilege to accelerate queries with them to the role.
```
Comment (evelynzhaojie) [inline docs/administration/User_privilege.md:431]: Grant SELECT privileges on all materialized views to the role to accelerate queries?

File: docs/administration/User_privilege.md [hunk 3]
```diff
@@ -454,62 +454,77 @@ We recommend you customize roles to manage privileges and users. The following e
...
 
-3. Grant privileges to perform backup and restore operations on global, database, table, and partition levels.
+3. Grant the read-only privilege on a specific external catalog:
+
+   ```SQL
+   --Create a role.
+   CREATE ROLE read_catalog_only;
+   --Switch to the corresponding catalog.
+   SET CATALOG hive_catalog;
+   --Grant SELECT privileges on all tables in all databases.
```
Comment (evelynzhaojie) [inline docs/administration/User_privilege.md:464]: use singular form "the SELETE privilege"

File: docs/administration/User_privilege.md [hunk 4]
```diff
@@ -420,15 +420,15 @@ We recommend you customize roles to manage privileges and users. The following e
 1. Grant global read-only privilege:
 
    ```SQL
-   -- Create a role.
+   --Create a role.
    CREATE ROLE read_only;
-   -- Grant USAGE privileges on all catalogs to the role.
+   --Grant USAGE privileges on all catalogs to the role.
```
Comment (evelynzhaojie) [inline docs/administration/User_privilege.md:425]: use singular form "the USAGE privilege"

File: docs/administration/User_privilege.md [hunk 5]
```diff
@@ -420,15 +420,15 @@ We recommend you customize roles to manage privileges and users. The following e
 1. Grant global read-only privilege:
```
Comment (evelynzhaojie) [inline docs/administration/User_privilege.md:420]: change global to system-level；中文原来写的“全局”，后来又改为“系统级”了，对应“权限项”里的级别，我们统一一下

File: docs/administration/User_privilege.md [hunk 6]
```diff
@@ -454,62 +454,77 @@ We recommend you customize roles to manage privileges and users. The following e
...
 
      ```SQL
-     --Create a role
+     --Create a role.
      CREATE ROLE recover_db;
-     --Grant the repository privilege on the system level
+     --Grant the REPOSITORY privilege on the SYSTEM level.
      GRANT REPOSITORY ON SYSTEM TO ROLE recover_db;
-     --Grant the privilege to create databases in the default catalog
+     --Grant the privilege to create databases.
```
Comment (evelynzhaojie) [inline docs/administration/User_privilege.md:496]: i prefer keeping "in the default catalog"

Comment (EsoragotoSpirit) [reply docs/administration/User_privilege.md:496]: 外部数据目录应该是不能在 StarRocks 内创建数据库的？这里默认语境就是 default catalog 了

**Review by evelynzhaojie — APPROVED** [review]

**Review by amber-create — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**